src/textual/layouts/horizontal.py

```diff
@@ -39,7 +39,9 @@ def arrange(
 
         x = box_models[0].margin.left if box_models else 0
 
-        for widget, box_model, margin in zip(parent.children, box_models, margins):
+        displayed_children = parent.displayed_children
```

Comment:
Contrary to `parent.children`, `parent.displayed_children` is a Python property getter; with this alias we'll trigger this property getter only once, and also makes the following code a wee bit easier to read